bokeh/charts/_charts.py

```diff
@@ -144,17 +147,44 @@ def start_plot(self, xgrid, ygrid):
         if ygrid:
             self.make_grid(1, yaxis.ticker)
 
-        # Add tools
+        # Add tools if supposed to
         if self.tools:
+            # need to add tool to all underlying plots
             for plot in self._plots:
+                removed_tools = []
+                # only add tools if the underlying plot hasn't been customized
+                # by some user injection
                 if not plot.tools:
-                    if not self.categorical:
-                        pan = PanTool()
-                        wheelzoom = WheelZoomTool()
-                        reset = ResetTool()
-                        plot.add_tools(pan, wheelzoom, reset)
-                    previewsave = PreviewSaveTool()
-                    plot.add_tools(previewsave)
+                    tools = []
+                    # if tools have been specified as string let's reuse
+                    # plotting machinery to create the related tools
+                    if isinstance(self.tools, string_types):
+                        for tool in re.split(r"\s*,\s*", self.tools.strip()):
+                            # re.split will return empty strings; ignore them.
+                            if tool == "":
+                                continue
```

Comment:
@bryevdv @damianavila I've added a more generic function to plotting_helpers. I've also added a test to check that plotting tools creation was getting the same results after refactoring but please take a look and the code to check if you spot something I overlooked...

A few things popped to me while working on this:

* was there a reason why plotting was using `self.tools.extend(tool_objs)` while charts used `plot.add_tools(*tool_objs)`. Any preference here? From an outsider look I prefer add_tools because it also provides some logic/validation...

* Shouldn't this just be a `Plot` method?